### PR TITLE
Add avifstreamtest for avifRWStream, avifROStream

### PR DESCRIPTION
**Pull request by y-guyon**

### CODE REVIEW

File: tests/gtest/avifstreamtest.cc [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+                               rw_full_box_version, rw_full_box_flags);
+
+  const uint16_t rw_someu16 = 2345;
+  avifRWStreamWriteU16(&rw_stream, rw_someu16);
+
+  avifRWStreamFinishBox(&rw_stream, rw_full_box_marker);
+
+  avifRWStreamFinishBox(&rw_stream, rw_box_marker);
+
+  const uint32_t rw_someu32 = 23456;
```
Comment (wantehchang) [inline tests/gtest/avifstreamtest.cc:54]: 23456 is equal to 0x5BA0, which fits in uint16_t also. It seems better to use a contant that won't fit in uint16_t. Similarly for 234567 and uint64_t in line 57.

File: tests/gtest/avifstreamtest.cc [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  avifRWStream rw_stream;
+  avifRWStreamStart(&rw_stream, &rw_data);
+  EXPECT_EQ(avifRWStreamOffset(&rw_stream), size_t{0});
+
+  const uint8_t rw_somedata[] = {3, 1, 4};
+  avifRWStreamWrite(&rw_stream, rw_somedata, sizeof(rw_somedata));
+
+  const char rw_somechars[] = "somechars";
+  avifRWStreamWriteChars(&rw_stream, rw_somechars,
+                         std::strlen(rw_somechars) + 1);
```
Comment (wantehchang) [inline tests/gtest/avifstreamtest.cc:32]: Nit: We can replace `std::strlen(rw_somechars) + 1` with `sizeof(rw_somechars)`.

File: tests/gtest/avifstreamtest.cc [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+  EXPECT_FALSE(avifROStreamHasBytesLeft(&ro_stream, rw_data.size + 1));
+  EXPECT_EQ(avifROStreamRemainingBytes(&ro_stream), rw_data.size);
+
+  std::vector<uint8_t> ro_somedata(sizeof(rw_somedata));
+  EXPECT_TRUE(
+      avifROStreamRead(&ro_stream, ro_somedata.data(), ro_somedata.size()));
+  EXPECT_TRUE(std::equal(rw_somedata, rw_somedata + sizeof(rw_somedata),
+                         ro_somedata.data()));
+
+  std::vector<char> ro_somechars(sizeof(rw_somechars) + 1);
```
Comment (wantehchang) [inline tests/gtest/avifstreamtest.cc:83]: `sizeof(rw_somechars)` already includes the terminating NUL character in `rw_somechars`, so it seems redundant to add 1 here.

**Review by wantehchang — COMMENTED** [review]
Yannis: Thank you for adding this test for `avifROStream` and `avifRWStream`. Please consider my review as supplemental.

**Review by maryla-uc — APPROVED** [review]
Thanks for adding tests!

File: tests/gtest/avifstreamtest.cc [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+  avifRWStreamWrite(&rw_stream, rw_somedata, sizeof(rw_somedata));
+
+  const char rw_somechars[] = "somechars";
+  avifRWStreamWriteChars(&rw_stream, rw_somechars, sizeof(rw_somechars));
+
+  const char rw_box_type[] = "type";
+  avifBoxMarker rw_box_marker =
+      avifRWStreamWriteBox(&rw_stream, rw_box_type, /*contentSize=*/0);
+
+  const uint8_t rw_someu8 = 0xA;
```
Comment (wantehchang) [inline tests/gtest/avifstreamtest.cc:36]: A hexadecimal digit is 4 bits, so `uint8_t` may have up to 2 nonzero hexadecimal digits. Similarly for the initializers for `rw_someu16`, etc.

**Pull request merged**